File: Agent14_Phase15_TestingQA_Completed/agent14-academic-risk-phase2/frontend/src/features/mentor/ai.ts
import api from "../../services/api";
import type { WhatIfRequest } from "./api";

export type CopilotIntent = "risk_summary" | "intervention_plan" | "what_if_explanation";

export interface CopilotResponse {
  student_id: string;
  intent: CopilotIntent;
  provider: string;
  model: string;
  grounded: boolean;
  explanation: string;
  recommended_actions: string[];
  priority_rationale: string;
  cautions: string[];
  source_snapshot: {
    risk_source: string;
    risk_types: string[];
    what_if_persistent?: boolean | null;
  };
}

export async function runMentorCopilot(studentId: string, intent: CopilotIntent, options?: {
  what_if?: WhatIfRequest;
  focus?: string;
}) {
  const response = await api.post<CopilotResponse>(`/mentor/ai/copilot/${studentId}`, {
    intent,
    what_if: options?.what_if,
    focus: options?.focus,
  });
  return response.data;
}

File: Agent14_Phase15_TestingQA_Completed/backend/app/schemas/ai.py
from __future__ import annotations

from enum import Enum
from typing import Optional
from pydantic import BaseModel, Field

from app.schemas.what_if import WhatIfRequest


class CopilotIntent(str, Enum):
    RISK_SUMMARY = "risk_summary"
    INTERVENTION_PLAN = "intervention_plan"
    WHAT_IF_EXPLANATION = "what_if_explanation"


class CopilotRequest(BaseModel):
    intent: CopilotIntent
    what_if: Optional[WhatIfRequest] = None
    focus: Optional[str] = Field(default=None, max_length=240)


class CopilotResponse(BaseModel):
    student_id: str
    intent: CopilotIntent
    provider: str
    model: str
    grounded: bool
    explanation: str
    recommended_actions: list[str]
    priority_rationale: str
    cautions: list[str]
    source_snapshot: dict
